lnbits/wallets/blink.py

```diff
@@ -0,0 +1,483 @@
+            ]
+
+            if not btc_wallet_ids:
+                raise ValueError("BTC Wallet not found")
+
+            self._wallet_id = btc_wallet_ids[0]
+            return self._wallet_id
+        except Exception as exc:
+            logger.warning(exc)
+            raise ValueError(f"Unable to connect to '{self.endpoint}'") from exc
+
+
+class BlinkGrafqlQueries(BaseModel):
+    balance_query: str
+    invoice_query: str
+    payment_query: str
+    status_query: str
+    wallet_query: str
+    tx_query: str
+    my_updates_query: str
+
+
+q = BlinkGrafqlQueries(
+    balance_query="""
+        query Me {
+          me {
+            defaultAccount {
+              wallets {
+                walletCurrency
+                balance
+              }
+            }
+          }
+        }
+        """,
+    invoice_query="""
+        mutation LnInvoiceCreateOnBehalfOfRecipient(
+          $input: LnInvoiceCreateOnBehalfOfRecipientInput!
+        ) {
+          lnInvoiceCreateOnBehalfOfRecipient(input: $input) {
+            invoice {
+              paymentRequest
+              paymentHash
+              paymentSecret
+              satoshis
+            }
+            errors {
+              message
+            }
+          }
+        }
+        """,
+    payment_query="""
+        mutation LnInvoicePaymentSend($input: LnInvoicePaymentInput!) {
+          lnInvoicePaymentSend(input: $input) {
+            status
+            errors {
+              message
+              path
+              code
+            }
+          }
+        }
+        """,
+    status_query="""
+        query InvoiceByPaymentHash($walletId: WalletId!, $paymentHash: PaymentHash!) {
+          me {
+            defaultAccount {
+              walletById(walletId: $walletId) {
+                invoiceByPaymentHash(paymentHash: $paymentHash) {
+                  ... on LnInvoice {
+                    paymentStatus
+                  }
+                }
+              }
+            }
+          }
+        }
+        """,
+    wallet_query="""
+        query me {
+          me {
+            defaultAccount {
+              wallets {
+                id
+                walletCurrency
+              }
+            }
+          }
+        }
+        """,
+    tx_query="""
+        query TransactionsByPaymentHash(
+          $walletId: WalletId!
+          $transactionsByPaymentHash: PaymentHash!
+        ) {
+          me {
+            defaultAccount {
+              walletById(walletId: $walletId) {
+                walletCurrency
+                ... on BTCWallet {
+                  transactionsByPaymentHash(paymentHash: $transactionsByPaymentHash) {
+                    settlementFee
+                    status
+                    direction
+                    settlementVia {
+                      ... on SettlementViaLn {
+                        preImage
+                      }
```

Comment:
The payments are not picked up when paying with Blink wallet or a Blink funded LNbits wallets which means that the intra-ledger payments are not picked up (not sure why only in LNcalendar).
The Blink backend does decide to settle depending on the invoice. It does return the preimage in this case also since it is known when creating the invoice.

`.... on SettlementViaIntraLedger` needs to be added to the query to cover for this case:

```suggestion
                    settlementVia {
                      ... on SettlementViaLn {
                        preImage
                      }
                      ... on SettlementViaIntraLedger {
                        preImage
                      }
```

Reply:
this change does not seem to be necessary as Blink intraledger payments are being picked  up in most extensions